contracts/btc-finality/src/finality.rs

```diff
@@ -396,8 +396,14 @@ pub fn handle_finality_signature(
     let fp_btc_pk_hex = &add_finality_sig.fp_btc_pk_hex;
     let height = add_finality_sig.height;
 
-    // Ensure the finality provider has voting power at this height
-    ensure_fp_has_power(deps.storage, height, fp_btc_pk_hex)?;
+    // Get the finality provider's voting power at this height (also ensures they have power)
+    let voting_power = get_fp_power(deps.storage, height, fp_btc_pk_hex)?;
+    if voting_power == 0 {
+        return Err(ContractError::NoVotingPower(
+            fp_btc_pk_hex.to_string(),
+            height,
+        ));
+    }
```

Comment:
`get_fp_power` will return error if no voting power so I'm not sure if we still want to keep this if check